### PR TITLE
Use on_pre_start callback

### PR DESCRIPTION
**Pull request by rwols**
Also, allow the user to set "command" in the .sublime-settings file. If it's present, use that instead.

### CODE REVIEW

**Comment by rchl** [comment]
I'm not at the computer right now but doesn't the command override work automatically without having to manually override anything?

**Comment by rwols** [comment]
not when it's already in the ClientConfig. Only sublime.Settings works like that.

File: plugin.py [hunk 1]
```diff
@@ -35,10 +37,10 @@ def name(cls) -> str:
...
 
     @classmethod
-    def configuration(cls) -> Tuple[sublime.Settings, str]:
-        settings, filename = super().configuration()
-        settings.set("command", cls.get_command())
-        return settings, filename
+    def get_settings(cls) -> sublime.Settings:
+        return sublime.load_settings(
+            "LSP-{}.sublime-settings".format(cls.name())
+        )
```
Comment (rchl) [inline plugin.py:43]: I don't know...

I'd instead remove `get_settings`, `get_command`, `on_pre_start` and `get_x_command()` and have the platform-specific parameters controlled through variables defined in `additional_variables()` (`binary_path` and `binary_arguments` or something like that).

Comment (rwols) [reply plugin.py:43]: but the command is completely different on Linux/macOS compared to Windows... not sure how you would unify that?

command on unix:

mono --assembly-loader=strict --config etc/config OmniSharp.exe --languageserver

command on Windows:

OmniSharp.exe --languageserver

Comment (rchl) [reply plugin.py:43]: One variable for `$server_binary` and one for `$server_arguments`?

Comment (rchl) [reply plugin.py:43]: ah, no, the arguments need to be in a list...

Comment (rchl) [reply plugin.py:43]: But then wouldn't it work to do it like now but decide and return correct command from the `configuration()` implementation already?

Comment (rwols) [reply plugin.py:43]: Possibly, but on_pre_start is exactly the callback we cooked up to set up the command at the last minute?

Comment (rchl) [reply plugin.py:43]: If the `configuration()` implementation is not the most appropriate place to return the configuration then I'm not sure what is. :)

The `on_pre_start` is for cases when we have to decide based on view/window or other context that it provides but in this case there is no need for that.

Comment (rwols) [reply plugin.py:43]: > If the configuration() implementation is not the most appropriate place to return the configuration then I'm not sure what is. :)

That's only for returning the settings object so LSP can add a settings change listener to it.

**Pull request merged**